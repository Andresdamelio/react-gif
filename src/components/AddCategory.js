import React, { useState } from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('');
  const [shoValidation, setShoValidation] = useState(false);
  const handleInputChange = (event) => {
    setInputValue(event.target.value);
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length > 2) {
      setCategories((categories) => [inputValue, ...categories]);
      setInputValue('');
      setShoValidation(false);
    } else {
      setShoValidation(true);
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        type='text'
        placeholder='Buscar gifs...'
        value={inputValue}
        onChange={handleInputChange}
      />
      {shoValidation && inputValue.trim().length < 3 && (
        <span className='error'>Debes ingresar al menos dos caracteres</span>
      )}
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
};

export default AddCategory;
