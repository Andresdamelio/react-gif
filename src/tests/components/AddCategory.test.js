import React from 'react';
import { shallow } from 'enzyme';
import AddCategory from '../../components/AddCategory';

describe('Probando componente <AddCategory />', () => {
  const setCategories = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test('Comprobando que el componente <AddCategory /> se renderiza de forma correcta', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Comprobando que evento en input de texto en  componente <AddCategory />', () => {
    const input = wrapper.find('input');
    input.simulate('change', { target: { value: 'Samurai' } });

    expect(wrapper.find('input').prop('value')).toBe('Samurai');
  });

  test('Mostrar validación si se ingresan menos de dos caracteres', () => {
    const form = wrapper.find('form');
    form.simulate('submit', { preventDefault: () => {} });
    const span = wrapper.find('span').text();

    expect(setCategories).not.toHaveBeenCalled();
    expect(span).toBe('Debes ingresar al menos dos caracteres');
  });

  test('Debe de llamar al setCategories y limpiar el input', () => {
    // 1. Simular cambio en input
    wrapper.find('input').simulate('change', { target: { value: 'Samurai' } });

    // 2. Simular submit
    wrapper.find('form').simulate('submit', { preventDefault: () => {} });

    // 3. Comprobar que se llama al setCategories
    expect(setCategories).toHaveBeenCalled();
    // Verificar que setCategories haya recibido una funcion
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));

    // 4. Comprobar que el input está vacío
    expect(wrapper.find('input').prop('value')).toBe('');
  });
});
