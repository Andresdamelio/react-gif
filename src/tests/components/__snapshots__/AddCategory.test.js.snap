// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando componente <AddCategory /> Comprobando que el componente <AddCategory /> se renderiza de forma correcta 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Buscar gifs..."
    type="text"
    value=""
  />
</form>
`;
